config/v1/types_cluster_version.go

```diff
@@ -88,9 +88,32 @@ type ClusterVersionSpec struct {
 	// +optional
 	Capabilities *ClusterVersionCapabilitiesSpec `json:"capabilities,omitempty"`
 
+	// signatureStores contains the upstream URIs to verify release signatures and optional
+	// reference to a config map by name containing the PEM-encoded CA bundle.
+	//
+	// By default, CVO will use existing signature stores if this property is empty.
+	// The CVO will check the release signatures in the local ConfigMaps first. It will search for a valid signature
+	// in these stores in parallel only when local ConfigMaps did not include a valid signature.
+	// Validation will fail if none of the signature stores reply with valid signature before timeout.
+	// Setting signatureStores will replace the default signature stores with custom signature stores.
+	// Default stores can be used with custom signature stores by adding them manually.
+	//
+	// A maximum of 32 signature stores may be configured.
+	// +kubebuilder:validation:MaxItems=32
+	// +openshift:enable:FeatureSets=TechPreviewNoUpgrade
+	// +listType=map
+	// +listMapKey=url
+	// +optional
+	SignatureStores []SignatureStore `json:"signatureStores"`
```

Comment:
Why not `omitempty`?

Reply:
Thanks, makes sense.

Reply:
> Current code in https://github.com/openshift/cluster-version-operator/pull/994 is distinguishing between "unset" (use the defaults) and "explicitly set to an empty array" (do not do any HTTP(S) signature store checking at all).

How are you doing this?

Typically a difference between an empty list and the list not being present is hard to explain and be understandable/usable to users so I'd veto that to be honest. It's much better to have an explicit option to allow the don't do any checking function, is that something you think we need?

This doesn't have omitempty because this is a configuration API, and we want the API to be discoverable. By rending the empty list it makes the API clear, there is an option here that I could configure if I wanted to. So IMO this should stay without the omitempty unless it is causing some serialisation issue that prevents it.